src/System.Net.Mail/tests/Functional/AlternateViewTest.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+//
+// AlternateViewTest.cs - NUnit Test Cases for System.Net.MailAddress.AlternateView
+//
+// Authors:
+//   John Luke (john.luke@gmail.com)
+//
+// (C) 2005 John Luke
+//
+
+using System;
+using System.IO;
+using System.Net.Mail;
+using System.Net.Mime;
+using Xunit;
+
+namespace System.Net.Mail.Tests
+{
+    public class AlternateViewTest
+    {
+        AlternateView av;
+
+        public AlternateViewTest()
+        {
+            av = AlternateView.CreateAlternateViewFromString("test", new ContentType("text/plain"));
+        }
+
+        [Fact]
+        public void ArgumentNullExceptionTest()
+        {
+            string s = null;
+            Assert.Throws<ArgumentNullException>(() => new AlternateView(s));
+        }
+
+        [Fact]
+        public void ArgumentNullException2Test()
+        {
+            Stream s = null;
+            Assert.Throws<ArgumentNullException>(() => new AlternateView(s));
+        }
+
+        [Fact]
+        public void ContentTypeTest()
+        {
+            Assert.NotNull(av.ContentType);
+            Assert.Equal("text/plain", av.ContentType.MediaType);
+        }
+
+        [Fact]
+        public void ContentType2Test()
+        {
+            AlternateView av = new AlternateView(new MemoryStream());
+            Assert.NotNull(av.ContentType);
+            Assert.Equal("application/octet-stream", av.ContentType.MediaType);
+        }
+
+        [Fact]
+        public void ContentStreamTest()
+        {
+            Assert.NotNull(av.ContentStream);
+            Assert.Equal(4, av.ContentStream.Length);
```

Comment:
This can be cleaned up later if desired.  It's strange to me that some of the tests are using an instance created in the ctor and others are creating their own instance.  I'd think it should be one or the other for consistency throughout the test class.